src/lib/timing-functions-shim.ts

```diff
@@ -266,44 +266,55 @@ export class TimingFunctionsShim implements TimingFunctions {
 
   /**
    * Provide a mechanism for running down queued animation frame callbacks.
+   * Since callbacks may schedule more animation frame callbacks, this method
+   * takes an optional frameCount parameter in indicate how many frames to
+   * advance.
+   *
+   * @param frameCount Number of frames to advance.
    */
-  runAnimationFrameCallbacks() {
-    // Remove all callbacks from the canonical animation frame callback queue.
-    // This local version will be the one we run down so that we can track any
-    // that were blocked by earlier exceptions. This also ensures that we don't
-    // accidentally start executing future queued callbacks (those put onto the
-    // canonical queue as a side effect of this run).
-    const presentCallbackQueue = this.animationFrameCallbackQueue.splice(
-        0, this.animationFrameCallbackQueue.length);
+  runAnimationFrameCallbacks(frameCount = 1) {
+    if (frameCount <= 0 || isNaN(frameCount) || !isFinite(frameCount)) {
```

Comment:
Why not use [`Number.isInteger()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger), provides better frame counting semantics and stronger type safety when used as follows.

```suggestion
    if (!Number.isInteger(frameCount) || frameCount <= 0) {
```

There is a difference in how `Number.isInteger()` [treats values like strings, `null`, etc.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger#using_isinteger) compared to the global `isFinite()`, but I think those semantics are what you intended here.